src/lib/DatePickerInput.tsx

```diff
@@ -82,27 +107,40 @@ const DatePickerInput = <T extends FieldValues>(props: DatePickerInputProps<T>)
     setValue(name, getConvertedDate(selectedDate));
   }, [name, selectedDate, setValue, getConvertedDate]);
 
+  const openDatePicker = useCallback(() => {
+    if (!isDisabled) {
+      internalDatePickerRef.current?.setOpen(true);
+    }
+  }, [isDisabled]);
+
+  const datePickerClassNames = classNames(className, "form-control", {
+    "border-start-0 rounded-0 rounded-end": !!addonLeft,
+    "border-end-0 rounded-0 rounded-start": !!addonRight,
+  });
+
   return (
     <Controller
       control={control}
       name={name}
       render={({ field, fieldState: { error } }) => (
-        <FormGroupLayout
+        <FormGroupLayout<T, DatePickerRenderAddonProps>
           helpText={helpText}
+          formGroupId={formGroupId.current}
           name={name}
           id={id}
           label={label}
           labelToolTip={labelToolTip}
           addonLeft={addonLeft}
           addonRight={addonRight}
-          inputGroupStyle={inputGroupStyle}
+          addonProps={{ openDatePicker, isDisabled }}
+          inputGroupStyle={!!addonLeft || !!addonRight ? { ...inputGroupStyle, alignItems: "normal" } : inputGroupStyle}
```

Comment:
@severinoAmmirati @manni497 

is it okay to always have align-items normal whenver there is an addon on the datepicker?

Reply:
@neotrow yes :D